pkg/menu/menu_test.go

```diff
@@ -0,0 +1,66 @@
+package menu
+
+import (
+	ui "github.com/gizak/termui/v3"
+	"strconv"
+	"testing"
+)
+
+func Test_newParagraph(t *testing.T) {
+	if err := ui.Init(); err != nil {
+		t.Fatal(err)
+	}
+	defer ui.Close()
+	const testText = "newParagraph test"
+	p := newParagraph(testText, false, 0, 50, 3)
+	if testText != p.Text {
+		t.Errorf("incorrect value for p.Text. got: %v, want: %v", p.Text, testText)
+	}
+}
+
+func pressKey(ch chan ui.Event, input []string) {
+	var key ui.Event
+	for _, id := range input {
+		key = ui.Event{
+			Type: ui.KeyboardEvent,
+			ID:   id,
+		}
+		ch <- key
+	}
+}
+
+func Test_GetInput(t *testing.T) {
+	if err := ui.Init(); err != nil {
+		t.Fatal(err)
+	}
+	defer ui.Close()
+	input := newParagraph("", true, 0, 50, 3)
+	warning := newParagraph("", false, 3, 50, 3)
+	testText := "100"
+	ch := make(chan ui.Event)
+	// mock user input
+	// first input is bad input "bad"
+	// second input is bad input "100a"
+	// third input is good input "100" but contain a process of typo then backspace
+	// now the warning text should be ""
+	go pressKey(ch, []string{"b", "a", "d", "<Enter>",
+		"1", "0", "0", "a", "<Enter>",
+		"1", "0", "a", "<Backspace>", "0", "<Enter>"})
+	isValid := func(input string) (bool, string) {
+		c, err := strconv.Atoi(input)
+		if err != nil || c < 0 {
+			return false, "Input is not a valid entry number."
+		}
+		return true, ""
+	}
+	inputText, err := processInput(input, warning, isValid, ch)
```

Comment:
Does it not work to use ui.PollEvents() to get the channel here instead of having to make your own function? That was the purpose of separating out the processInput function right?

Reply:
yes, I tried. it works too